### ISSUE TITLE
feat: Should apisix dashboard support multi-tenancy feature?

### ISSUE THREAD

**Issue by gy09535** [apache/apisix-dashboard#648]
Sometimes  we can split apis to different  apisix cluster for reduce risk , dashboard can manage more then one   apisix cluster. 

- Why do you submit this issue?
- [x] Question or discussion
- [ ] Bug
- [x] Requirements
- [x] Feature or performance improvement
- [ ] Other

___
### Question
- What do you want to know?

___
### Bug
- Which version of Apache APISIX Dashboard, OS, and Browser?

- What happened?
If possible, provide a way to reproduce the error.

___
### Requirements or improvement
- Please describe your requirements or improvement suggestions.

**Comment by juzhiyuan** [comment]
Ya This is a good feature, we may support it one day. BTW, would you have interests to draft a proposal about how to implement this feature?

**Comment by juzhiyuan** [comment]
cc @membphis @moonming @ShiningRush

**Comment by moonming** [comment]
@gy09535 I think you want to manager Apache APISIX clusters in one dashboard, right?

**Comment by gy09535** [comment]
> @gy09535 I think you want to manager Apache APISIX clusters in one dashboard, right?

yeah

**Comment by moonming** [comment]
@gxthrj @spacewander what do you think?

**Comment by ShiningRush** [comment]
I think this is a useful feature, such as using a dashboard to manage API gateways and streaming gateways, but the priority can be lowered.
Because usually the number of gateway clusters is not too much, even if per gateway uses a dashboard, it will not bring too much burden to users.
BTW: The reason why the k8s cluster needs federation here is because there are indeed many clusters, especially for cloud platforms

**Comment by gy09535** [comment]
In my environment  has more the 2000 apis , some apis is for transaction, they are core function , we should split it in different cluster,  cluster count is related with your business.

**Comment by ShiningRush** [comment]
I think that the measurement of route should not be based on the api, but the service. Generally speaking, a service matches a route. We will not generate a separate route for each API, which is difficult to maintain.
Then there is another solution that APISIX deploys multiple clusters, and then directly use same dashboard to manage them (in fact, we do this internally). The disadvantage of this approach is that the routes will mix together, but if there is a nice filtering in dashboard, it is actually also acceptable.

**Comment by juzhiyuan** [comment]
looking forward to more better practices haha

**Comment by gy09535** [comment]
> I think that the measurement of route should not be based on the api, but the service. Generally speaking, a service matches a route. We will not generate a separate route for each API, which is difficult to maintain.
> Then there is another solution to handle your case that APISIX deploys multiple clusters, and then directly use same dashboard to manage them (in fact, we do this internally). The disadvantage of this approach is that the routes will mix together, but if there is a nice filtering in dashboard, it is actually also acceptable.

yeah, we can create  different cluster manager in same dashboard, for business gateway , how to set the route is a good question it relate with how  manage apis , I want to make it as a micro service gateway , and I can sync api from service with swagger , in this way api config is easy. If a group api has the same feature ,may be we can set it as wildcard.

**Comment by gxthrj** [comment]
Yes, I think it is a useful. The dashboard should be only one .
In [ingress controller](https://github.com/api7/ingress-controller) , we use `ingress.class` to distinguish different clusters.

**Comment by juzhiyuan** [comment]
conclusion?

**Comment by bzp2010** [comment]
There is still no relevant contribution and no mature discussed solution, you can participate in the planning of the V3 version currently under discussion and perhaps support it later.

DISCUSS email: [https://lists.apache.org/thread/lltjvf2bjj18og6no5lplwjry0z01pbd](https://lists.apache.org/thread/lltjvf2bjj18og6no5lplwjry0z01pbd)

**Comment by zambardus** [comment]
If this is not supported out-of-box, Can someone please guide me on how can I achieve multi-tenancy by making changes at my end?

**Comment by bzp2010** [comment]
Based on the principle of mapping to the Admin API, this will not be implemented, and UI access restrictions or authentication will reuse the APISIX mechanism. ref https://github.com/apache/apisix-dashboard/issues/2981

Conceptually, multi-tenancy would equal multiple APISIX clusters using different etcd clusters, or different etcd prefixes.

If you have any questions, please start a new issue!
